ddtrace/propagation/http.py

```diff
@@ -483,10 +489,201 @@ def _extract(headers):
+      trace-id        = 32HEXDIGLC
+      parent-id       = 16HEXDIGLC
+      trace-flags     = 2HEXDIGLC
+
+    Example value of HTTP ``traceparent`` header::
+        value = 00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01
+        base16(version) = 00
+        base16(trace-id) = 4bf92f3577b34da6a3ce929d0e0e4736
+        base16(parent-id) = 00f067aa0ba902b7
+        base16(trace-flags) = 01  // sampled
+
+    The format for ``tracestate`` is key value pairs with each entry limited to 256 characters.
+    An example of the ``dd`` list member we would add is::
+    "dd=s:2;o:rum;t.dm:-4;t.usr.id:baz64"
+
+    Implementation details:
+      - Datadog Trace and Span IDs are 64-bit unsigned integers.
+      - The W3C Trace Context Trace ID is a 16-byte hexadecimal string.
+      - If the incoming traceparent was valid we DO NOT update the trace-id.
+        Otherwise, the value is set to the hex-encoded value of the trace-id.
+        If the trace-id is a 64-bit value (i.e. a Datadog trace-id),
+        then the upper half of the hex-encoded value will be all zeroes.
+
+      - The tracestate header will have one list member added to it, ``dd``, which contains
+        values that would be in x-datadog-tags as well as those needed for propagation information.
+        The keys to the ``dd`` values have been shortened as follows to save space:
+        ``sampling_priority`` = ``s``
+        ``origin`` = ``o``
+        ``_dd.p.`` prefix = ``t.``
+    """
+
+    @staticmethod
+    def _is_valid_datadog_trace_tag_key(key):
+        return key.startswith("t.")
+
+    @staticmethod
+    def _get_traceparent_values(headers):
+        # type: (Dict[str, str]) -> Optional[Tuple[str,int,int,int]]
+
+        tp = _extract_header_value(_POSSIBLE_HTTP_HEADER_TRACEPARENT, headers)
+        if tp is None:
+            return None
+
+        try:
+            version, trace_id_hex, span_id_hex, trace_flags_hex = tp.split("-")
+            # check version is a valid hexadecimal
+            int(version, 16)
+            # currently 00 is the only version format, but if future versions come up we may need to add changes
+            if version != "00":
+                log.warning("unsupported traceparent version:%s . Will still attempt to parse.", (version))
+
+            if len(trace_id_hex) == 32 and len(span_id_hex) == 16 and len(trace_flags_hex) >= 2:
+                trace_id = _hex_id_to_dd_id(trace_id_hex)
+                span_id = _hex_id_to_dd_id(span_id_hex)
+
+                # All 0s are invalid values
+                assert trace_id != 0
+                assert span_id != 0
+
+                trace_flags = _hex_id_to_dd_id(trace_flags_hex)
+                # there's currently only one trace flag, which denotes sampling priority
+                # was set to keep "01" or drop "00"
+                # trace flags is a bit field: https://www.w3.org/TR/trace-context/#trace-flags
+                sampling_priority = int(trace_flags & 0x1)
```

Comment:
```suggestion
                sampling_priority = trace_flags & 0x1
```

the type of `<int> & <int>` is `<int>` so we shouldn't need an additional `int()` call here.